core/src/main/java/amino/run/kernel/server/KernelServerImpl.java

```diff
@@ -392,6 +403,18 @@ public MemoryStatThread getMemoryStatThread() {
         return new MemoryStatThread();
     }
 
+    /**
+     * Receive HeartBeat with random bytes to measure data transfer time
+     *
+     * @param data
+     */
+    @Override
+    public void receiveHeartBeat(RandomData data) {}
```

Comment:
Come to think of it, the compiler/JIT might optimize this data away because it's never used.  Might be worth checking that.  I'm not sure what the best solution is if it is being optimized away.  You might need to access it somehow, without using up too many CPU cycles.

Reply:
Have checked it. Call is not optimized. Probably because it is an RPC.